include/zephyr/posix/posix_features.h

```diff
@@ -267,42 +267,33 @@ __PICOLIBC_MINOR__ == 8 && __PICOLIBC_PATCHLEVEL__ >= 9)))
 #define _POSIX_AIO_MAX                      (1)
 #define _POSIX_ARG_MAX                      (4096)
 #define _POSIX_CHILD_MAX                    (25)
-#define _POSIX_DELAYTIMER_MAX \
-	COND_CODE_1(CONFIG_POSIX_TIMERS, (CONFIG_POSIX_DELAYTIMER_MAX), (0))
-#define _POSIX_HOST_NAME_MAX \
-	COND_CODE_1(CONFIG_POSIX_NETWORKING, (CONFIG_POSIX_HOST_NAME_MAX), (0))
+#define _POSIX_DELAYTIMER_MAX               (32)
+#define _POSIX_HOST_NAME_MAX                (255)
```

Comment:
Not sure how we handle the conflicting max host name lengths in various places. Currently networking sets hostname length to max. 63 for dynamic hostname updates (there is buffer allocated by this length). For static (built time configured) hostname there is no limit.
This `_POSIX_HOST_NAME_MAX` value might not be related to what networking is setting. What is this value used for in the first place? Should we use this value in networking side if posix is enabled?

Reply:
`_POSIX_HOST_NAME_MAX` is a minimum value (constant, never changes).

`HOST_NAME_MAX` is runtime invariant (compile-time constant, is supposed to be larger than the minimum value).

`HOST_NAME_MAX` should be set to `CONFIG_POSIX_HOST_NAME_MAX` so that it's configurable and the Kconfig option should mirror (default to, but not be user configurable) what's in the network Kconfig.

Of course, there is also `sysconf(_SC_HOST_NAME_MAX)` or something like that for runtime queries.